lib/java-extras/src/test/java/org/triplea/io/ZipExtractorTest.java

```diff
@@ -25,42 +48,42 @@
-      final File testDataZip = testDataZip();
-
-      ZipExtractor.unzipFile(testDataZip, destinationFolder);
-
-      assertThat(
-          "Should contain one folder and one file only",
-          FileUtils.listFiles(destinationFolder.toPath()),
-          hasSize(2));
-
-      final File unzippedTestFile1 = new File(destinationFolder, "test-file1");
-      final String testFile1Contents = Files.readString(unzippedTestFile1.toPath());
-      assertThat(testFile1Contents, is("ipsum ipsor\n"));
-
-      final File unzippedDirectory = new File(destinationFolder, "directory");
-      assertThat(unzippedDirectory.isDirectory(), is(true));
-      assertThat(unzippedDirectory.exists(), is(true));
-
-      final File unzippedTestFile2 = new File(unzippedDirectory, "test-file2");
-      final String testFile2Contents = Files.readString(unzippedTestFile2.toPath());
-      assertThat(testFile2Contents, is("lupsom osculus\n"));
-    } finally {
-      deleteDirectory(destinationFolder);
-    }
+    final Path testDataZip = testDataFolder.resolve("test-data.zip");
+
+    ZipExtractor.unzipFile(testDataZip, destinationFolder);
+
     assertThat(
-        "Ensure that we have no file handles open or anything else that"
-            + "would get in the way of deleting the unzipped folder",
-        destinationFolder.exists(),
-        is(false));
+        "Should contain one folder and one file only",
+        FileUtils.listFiles(destinationFolder),
+        hasSize(2));
+
+    final Path unzippedTestFile1 = destinationFolder.resolve("test-file1");
+    final String testFile1Contents = Files.readString(unzippedTestFile1);
+    assertThat(testFile1Contents, is("ipsum ipsor\n"));
+
+    final Path unzippedDirectory = destinationFolder.resolve("directory");
+    assertThat(Files.isDirectory(unzippedDirectory), is(true));
+    assertThat(Files.exists(unzippedDirectory), is(true));
+
+    final Path unzippedTestFile2 = unzippedDirectory.resolve("test-file2");
+    final String testFile2Contents = Files.readString(unzippedTestFile2);
+    assertThat(testFile2Contents, is("lupsom osculus\n"));
   }
 
-  private static File testDataZip() {
-    final File file =
-        new File(ZipExtractorTest.class.getClassLoader().getResource("test-data").getFile());
-    return new File(file, "test-data.zip");
+  @Test
+  void verifyMaliciousZipCantBeUnpacked() {
+    final Path zip = testDataFolder.resolve("evil.zip");
+    final Path subfolder = destinationFolder.resolve("sub");
+
+    final Exception exception =
+        assertThrows(
+            ZipExtractor.ZipSecurityException.class, () -> ZipExtractor.unzipFile(zip, subfolder));
+
+    assertThat(Files.exists(destinationFolder.resolve("matrix.jpg")), is(false));
+    // Make sure file isn't extracted at all
+    assertThat(Files.exists(subfolder.resolve("matrix.jpg")), is(false));
+
+    // Folder creation outside of the extraction directory should be prevented!
+    assertThat(exception.getMessage(), is("/.."));
```

Comment:
Is "/.." really *the* exception message? That seems like not much for an exception message.
Second comment, testing exact message exceptions is brittle. Generally better to check that an exception contains key pieces of information. IN this case, do we even need to check anything on the exception at all really other than it was thrown?

Reply:
It's not that bad when it is printed as a Stacktrace:
```
ZipSecurityException: /..
```

but I get the point you're making